src/Generator/Passes/MarshalStructAsValTypeIfPossiblePass.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using CppSharp.AST;
+
+namespace CppSharp.Passes
+{
+    public class MarshalStructAsValTypeIfPossiblePass : TranslationUnitPass
```

Comment:
I think we should be more consistent here and rename this to something like our existing passes, maybe `CheckValueTypeClassesPass` or something.